### PR TITLE
Renaming base classes

### LINKED ISSUE #921: Base Classes aren't named consistently
There is `AnalysisBase`, `Reader`, `ProtoReader` and I'm sure some that I forgot in the topology system and other places. Almost pure Abstract classes like `AnalysisBase` should follow a consistent naming scheme.

Personally I would like `BaseXXX` because my fingers always want to type it this pattern but `XXXBase` is just as fine.
# List of abstract base classes
- AnalysisBase
- ProtoReader
- Reader

### PR DESCRIPTION
**Pull request by richardjgowers**
Renamed various base classes:

IObase -> IOBase
Reader -> ReaderBase
SingleFrameReader -> SingleFrameReaderBase
Writer -> WriterBase
SelectionWriter -> SelectionWriterBase
TopologyReader -> TopologyReaderBase

Fixes #921

PR Checklist
------------
 - [ ] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

File: package/MDAnalysis/analysis/leaflet.py [hunk 1]
```diff
@@ -114,10 +117,10 @@ def __init__(self, universe, selectionstring, cutoff=15.0, pbc=False, sparse=Non
                  use fast :func:`~MDAnalysis.analysis.distances.distance_array`
                  implementation [``None``].
         """
-        universe = MDAnalysis.as_Universe(universe)
+        universe = universe.as_Universe(universe)
```
Comment (mnmelo) [inline package/MDAnalysis/analysis/leaflet.py]: I don't think it's good practice to clobber the imported `universe` name.

File: package/doc/sphinx/source/documentation_pages/selections_modules.rst [hunk 2]
```diff
@@ -12,7 +12,7 @@ The classes in this module allow writing a selection to a file that can be read
 allows a user to combine their favourite tools with MDAnalysis for further
 visualization or simulation.
 
-There exist different :class:`~MDAnalysis.selections.base.SelectionWriter` classes
+There exist different :class:`~MDAnalysis.selections.base.SelectionWriterBase` classes
 for different package. The :func:`~MDAnalysis.selections.base.get_writer` function
```
Comment (mnmelo) [inline package/doc/sphinx/source/documentation_pages/selections_modules.rst]: While we're at it, could you fix the 'packages' (plural) typo?

**Comment by orbeckst** [comment]
@mnmelo can you have a last quick look and then merge?

**Comment by richardjgowers** [comment]
@orbeckst Proto isn't immediately useful as a Base, so I didn't want to label it in the same way

**Pull request merged**